SYCL/ESIMD/api/functional/ctors/ctor_default.cpp

```diff
@@ -116,44 +116,19 @@ template <typename DataT, int NumElems, typename TestCaseT> struct test {
   }
 };
 
-template <typename DataT, typename TestT>
-using run_test_with_one_elem = test<DataT, 1, TestT>;
-
-template <typename DataT, typename TestT>
-using run_test_with_eight_elems = test<DataT, 8, TestT>;
-
-template <typename DataT, typename TestT>
-using run_test_with_sixteen_elems = test<DataT, 16, TestT>;
-
-template <typename DataT, typename TestT>
-using run_test_with_thirty_two_elems = test<DataT, 32, TestT>;
-
-template <typename TestT, typename... T>
-bool run_verification_with_chosen_test_type(
-    sycl::queue &queue, const named_type_pack<T...> &types) {
-  bool passed{true};
-
-  passed &= for_all_types<run_test_with_one_elem, TestT>(types, queue);
-  passed &= for_all_types<run_test_with_eight_elems, TestT>(types, queue);
-  passed &= for_all_types<run_test_with_sixteen_elems, TestT>(types, queue);
-  passed &= for_all_types<run_test_with_thirty_two_elems, TestT>(types, queue);
-  return passed;
-}
-
 int main(int argc, char **argv) {
-  sycl::queue queue{esimd_test::ESIMDSelector{},
-                    esimd_test::createExceptionHandler()};
+  sycl::queue q{esimd_test::ESIMDSelector{},
+                esimd_test::createExceptionHandler()};
 
   bool passed{true};
 
   const auto types{get_tested_types<tested_types::all>()};
+  const auto vector_lengths{get_all_vector_lengths()};
 
-  passed &= run_verification_with_chosen_test_type<initializer>(queue, types);
-  passed &=
-      run_verification_with_chosen_test_type<var_declaration>(queue, types);
-  passed &=
-      run_verification_with_chosen_test_type<rval_in_expression>(queue, types);
-  passed &= run_verification_with_chosen_test_type<const_ref>(queue, types);
+  passed &= for_types_and_dims<test, initializer>(types, vector_lengths, q);
```

Comment:
```suggestion
  passed &= for_types_and_dims<test, initializer>(types, dims, q);
```
Let's use a solid naming over the tests, with no synonyms for the same term. So if you have a `dimensions` term defined by using `for_all_types_and_dims`, let's use it over the code.

Reply:
I've applied this comment [here](https://github.com/intel/llvm-test-suite/pull/588/commits/f4c3940c7788727dae45410460dde6655274266d#diff-6d44d6893e30c56cec1c2e698ea43849081d8b092f7a87532bd567294d14ed69R120)